src/components/WorkOrder/WorkOrderDetails.js

```diff
@@ -22,48 +20,39 @@ const WorkOrderDetails = ({
 }) => {
   const { user } = useContext(UserContext)
 
+  const allowedMenu = () => {
+    return WORK_ORDER_DROP_DOWN_MENU.filter((choice) => {
+      if (
+        choice.permittedRoles.some((role) => user.roles.includes(role)) &&
+        choice.permittedStatus.includes(workOrder.status)
+      ) {
+        return choice
+      }
+    })
+  }
+
   return (
-    <div>
+    <>
       <BackButton />
-      <div>
-        <h1 className="lbh-heading-l display-inline govuk-!-margin-right-6">
-          Works order: {workOrder.reference}
-        </h1>
-        {user &&
-          (user.hasContractorPermissions ||
-            user.hasContractManagerPermissions) && (
-            <Link href={`/work-orders/${workOrder.reference}/choose-option`}>
-              <a className="govuk-body-m">Update Works Order</a>
-            </Link>
-          )}
-        {user &&
-          user.hasContractManagerPermissions &&
-          workOrder.status ===
-            STATUS_VARIATION_PENDING_APPROVAL.description && (
-            <Link
-              href={`/work-orders/${workOrder.reference}/variation-authorisation`}
-            >
-              <a className="govuk-body-m">Authorise Works Order Variation</a>
-            </Link>
-          )}
-        {user &&
-          user.hasAuthorisationManagerPermissions &&
-          workOrder.status ===
-            STATUS_AUTHORISATION_PENDING_APPROVAL.description && (
-            <Link href={`/work-orders/${workOrder.reference}/authorisation`}>
-              <a className="govuk-body-m">Authorise Works Order</a>
-            </Link>
-          )}
-        {user &&
-          (user.hasAgentPermissions ||
-            user.hasAuthorisationManagerPermissions ||
-            user.hasContractManagerPermissions) &&
-          workOrder.status !== STATUS_CANCELLED.description && (
-            <Link href={`/work-orders/${workOrder.reference}/cancel`}>
-              <a className="govuk-body-m">Cancel Works Order</a>
-            </Link>
+
+      <GridRow>
+        <GridColumn width="two-thirds">
+          <h1 className="lbh-heading-l display-inline govuk-!-margin-right-6">
+            Works order: {workOrder.reference}
+          </h1>
+        </GridColumn>
+        <GridColumn width="one-third">
+          {allowedMenu()?.length > 0 && (
+            <MultiButton
+              name="workOrderMenu"
+              label="Select work order"
+              secondary={false}
+              choices={allowedMenu()}
```

Comment:
We're calculating the allowed menu actions twice if the first passes the condition - once on this line and once on line 45. Can we get away with calculating them when the component loads once and then referring to that variable? E.g. assign `allowedMenu` to the result of the filtering instead of a function.